tests/platform_tests/test_platform_info.py

```diff
@@ -292,39 +282,49 @@ def test_turn_on_off_psu_and_check_psustatus(duthosts,
         all_outlet_status = all_outlet_status[0:-2]
         logging.info(
             "DUT is MgmtTsToR, the last 2 outlets are reserved for Console Switch and are not visible from DUT.")
-    for outlet in all_outlet_status:
-        psu_under_test = None
-        if outlet['outlet_on'] is False:
-            continue
 
-        logging.info("Turn off outlet {}".format(outlet))
-        pdu_ctrl.turn_off_outlet(outlet)
-        time.sleep(PDU_WAIT_TIME)
+    # Group outlets/PDUs by PSU and toggle PDUs by PSU
+    psu_to_pdus = get_grouped_pdus_by_psu(pdu_ctrl)
 
-        cli_psu_status = duthost.command(CMD_PLATFORM_PSUSTATUS)
-        for line in cli_psu_status["stdout_lines"][2:]:
-            psu_match = psu_line_pattern.match(line)
-            pytest_assert(psu_match, "Unexpected PSU status output")
-            # also make sure psustatus is not 'NOT PRESENT', which cannot be turned on/off
-            if psu_match.group(2) != "OK" and psu_match.group(2) != "NOT PRESENT":
-                psu_under_test = psu_match.group(1)
-            check_vendor_specific_psustatus(duthost, line, psu_line_pattern)
-        pytest_assert(psu_under_test is not None, "No PSU is turned off")
-
-        logging.info("Turn on outlet {}".format(outlet))
-        pdu_ctrl.turn_on_outlet(outlet)
-        time.sleep(PDU_WAIT_TIME)
-
-        cli_psu_status = duthost.command(CMD_PLATFORM_PSUSTATUS)
-        for line in cli_psu_status["stdout_lines"][2:]:
-            psu_match = psu_line_pattern.match(line)
-            pytest_assert(psu_match, "Unexpected PSU status output")
-            if psu_match.group(1) == psu_under_test:
-                pytest_assert(psu_match.group(2) == "OK",
-                              "Unexpected PSU status after turned it on")
-            check_vendor_specific_psustatus(duthost, line, psu_line_pattern)
-
-        psu_test_results[psu_under_test] = True
+    try:
+        for psu in psu_to_pdus.keys():
+            outlets = psu_to_pdus[psu]
+            psu_under_test = None
+
+            logging.info("Turning off {} PDUs connected to {}".format(len(outlets), psu))
+            for outlet in outlets:
+                pdu_ctrl.turn_off_outlet(outlet)
+            time.sleep(PDU_WAIT_TIME)
```

Comment:
Can we separate the sleep time on chassis and non-chassis devices? For example, we can do something like this:
![image](https://github.com/user-attachments/assets/cd3723ac-e055-4b67-8623-1401bba403fe)

Reply:
Updated in latest commit, please check :)